File: code/grpc/code.go
package grpc2

import (
	codev1 "Webook/api/proto/gen/api/proto/code/v1"
	"Webook/code/service"
	"context"
	"google.golang.org/grpc"
)

type CodeServiceServer struct {
	service service.CodeService
	codev1.UnimplementedCodeServiceServer
}

func NewCodeServiceServer(svc service.CodeService) *CodeServiceServer {
	return &CodeServiceServer{
		service: svc,
	}
}
func (c *CodeServiceServer) Register(server grpc.ServiceRegistrar) {
	codev1.RegisterCodeServiceServer(server, c)
}
func (c *CodeServiceServer) Send(ctx context.Context, req *codev1.CodeSendRequest) (*codev1.CodeSendResponse, error) {
	err := c.service.Send(ctx, req.Biz, req.Phone)
	return &codev1.CodeSendResponse{}, err
}

func (c *CodeServiceServer) Verify(ctx context.Context, req *codev1.VerifyRequest) (*codev1.VerifyResponse, error) {
	ans, err := c.service.Verify(ctx, req.Biz, req.Phone, req.InputCode)
	return &codev1.VerifyResponse{
		Answer: ans,
	}, err
}

File: frontEnd/webook-frontend/src/api/code.ts
import { get, post } from './http'

// 验证码相关接口类型定义
export interface CodeSendRequest {
  biz: string
  phone: string
}

export interface VerifyRequest {
  biz: string
  phone: string
  inputCode: string
}

export interface VerifyResponse {
  answer: boolean
}

// 验证码相关API
export const codeApi = {
  // 发送验证码
  sendCode: (data: CodeSendRequest) => {
    return post('/code/send', data)
  },
  
  // 验证验证码
  verifyCode: (data: VerifyRequest) => {
    return post<VerifyResponse>('/code/verify', data)
  }
}

